crates/server/tests/handlers/get_pricing_data_tests.rs

```diff
@@ -0,0 +1,180 @@
+use axum::{
+    extract::State,
+    http::StatusCode,
+    Json,
+};
+use db_access::{
+    models::JobStatus,
+    queries::{create_job_request, get_job_request, update_job_status},
+    DbConnection,
+};
+use mockall::predicate::*;
+use server::handlers::get_pricing_data::{get_pricing_data, PitchLakeJobRequest, PitchLakeJobRequestParams};
+
+// Mock the database connection
+mock! {
+    pub DbPool {}
+    impl DbConnection for DbPool {
+        fn pool(&self) -> &sqlx::PgPool;
+    }
+}
+
+// Helper function to create a mock DB connection
+fn mock_db_connection() -> DbConnection {
+    let mut mock_pool = MockDbPool::new();
+    mock_pool.expect_pool().returning(|| unimplemented!());
+    DbConnection::new(mock_pool)
```

Comment:
Imo if we are mocking the db, it should be a unit tests and be placed under the handlers as such. The `tests` directory should mainly be reserved for integration or e2e tests.

Reply:
This is more of a `nit`, so we can always refactor later.

Reply:
I'm not used to having unit tests in the same file as the source code, but I'm flexible. We can do that